model/src/main/java/com/epam/esm/model/parameters/CertificateSearchParameter.java

```diff
@@ -0,0 +1,20 @@
+package com.epam.esm.model.parameters;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum CertificateSearchParameter {
+    TAG_NAME(CertificateParameter.TAG_NAME),
+    CERTIFICATE_NAME(CertificateParameter.CERTIFICATE_NAME),
+    CERTIFICATE_DESCRIPTION(CertificateParameter.CERTIFICATE_DESCRIPTION);
+    CertificateParameter certificateParameter;
+
+    CertificateSearchParameter(CertificateParameter certificateParameter) {
+        this.certificateParameter = certificateParameter;
+    }
+    public static Optional<CertificateSearchParameter> getByType(String type) {
```

Comment:
Please, review `Optional` usage in your project and use it if it is necessary

Reply:
done